metrics/web/test/features/dashboard/presentation/widgets/metrics_table_test.dart

```diff
@@ -0,0 +1,232 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:metrics/features/common/presentation/metrics_theme/store/theme_store.dart';
+import 'package:metrics/features/common/presentation/metrics_theme/widgets/metrics_theme_builder.dart';
+import 'package:metrics/features/dashboard/presentation/model/project_metrics_data.dart';
+import 'package:metrics/features/dashboard/presentation/state/project_metrics_store.dart';
+import 'package:metrics/features/dashboard/presentation/strings/dashboard_strings.dart';
+import 'package:metrics/features/dashboard/presentation/widgets/build_number_text_metric.dart';
+import 'package:metrics/features/dashboard/presentation/widgets/circle_percentage.dart';
+import 'package:metrics/features/dashboard/presentation/widgets/metrics_table.dart';
+import 'package:metrics/features/dashboard/presentation/widgets/metrics_table_header.dart';
+import 'package:metrics/features/dashboard/presentation/widgets/project_metrics_tile.dart';
+import 'package:metrics/features/dashboard/presentation/widgets/sparkline_graph.dart';
+import 'package:states_rebuilder/states_rebuilder.dart';
+
+import '../test_utils/project_metrics_store_stub.dart';
+
+void main() {
+  group("MetricsTable", () {
+    testWidgets(
+      "contains MetricsTitle widget",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(_MetricsTableTestbed());
+
+        expect(find.byType(MetricsTableHeader), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "contains ListView with ProjectMetricTile widgets",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(_MetricsTableTestbed());
+        await tester.pumpAndSettle();
+
+        expect(
+          find.descendant(
+            of: find.byType(ListView),
+            matching: find.byType(ProjectMetricsTile),
+            matchRoot: true,
+          ),
+          findsOneWidget,
+        );
+      },
+    );
+
+    testWidgets(
+      "displays an error, occured during loading the metrics data",
+      (WidgetTester tester) async {
+        const metricsStore = _MetricsStoreErrorStub();
+
+        await tester.pumpWidget(_MetricsTableTestbed(
+          metricsStore: metricsStore,
+        ));
+
+        await tester.pumpAndSettle();
+
+        expect(
+          find.text(DashboardStrings.getLoadingErrorMessage(
+              '${_MetricsStoreErrorStub.errorMessage}')),
```

Comment:
Maybe move this `getLoadingErrorMessage` method calling to the separate variable to improve readability?